src/semantic_digital_twin/adapters/ros/messages.py

```diff
@@ -178,23 +174,25 @@ class WorldModelSnapshot(SubclassJSONSerializer):
     def to_json(self) -> Dict[str, Any]:
         return {
             **super().to_json(),
-            "modifications": [m.to_json() for m in self.modifications],
+            "modifications": to_json(self.modifications),
             "state": {
-                "prefixed_names": [n.to_json() for n in self.prefixed_names],
-                "states": list(self.states),
+                "prefixed_names": to_json(self.prefixed_names),
+                "states": to_json(list(self.states)),
             },
         }
 
     @classmethod
     def _from_json(cls, data: Dict[str, Any], **kwargs) -> Self:
         state = data.get("state", {})
+
+        # Support both legacy top-level fields and the nested 'state' structure
```

Comment:
why both?

Reply:
which test? is it a hardcoded one? i don't see the point in supporting the old to_json implementation. shouldn't all jsons be produced be the same version of semdt?